tests/unit/test_xasprintf_x.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-FileCopyrightText: 2023, Alejandro Colomar <alx@kernel.org>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+
+#include <setjmp.h>
+#include <stddef.h>
+#include <stdlib.h>
+
+#include <stdarg.h>  // Required by <cmocka.h>
+#include <stddef.h>  // Required by <cmocka.h>
+#include <setjmp.h>  // Required by <cmocka.h>
+#include <stdint.h>  // Required by <cmocka.h>
+#include <cmocka.h>
+
+#include "sprintf.h"
+
+
+#define assert_unreachable()  assert_true(0)
+
+
+static jmp_buf  jmpb;
+
+
+/**********************
+ * WRAPPERS
+ **********************/
+int
+__wrap_vasprintf(char **restrict p, const char *restrict fmt, va_list ap)
+{
+	return -1;
```

Comment:
You can reduce the number of files needed to test this function by changing this line to `return mock();`. Then, in the test cases (`test_xasprintf_x` and `test_xasprintf`) you need to add the mock() function return value with `will_return(__wrap_vasprintf, -1);` (in the case of `test_xasprintf` it would be the length of `foo1bar`)

Reply:
I did something different.  Please check if you like it.  (I'll push in a moment, after testing.)